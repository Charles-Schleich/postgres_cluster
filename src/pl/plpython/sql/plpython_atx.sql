CREATE EXTENSION plpythonu;

create table atx_test(a text, b int);

create or replace function pythonomous() returns void as $$
	plpy.execute("insert into atx_test values ('asd', 123)")

	try:
		with plpy.autonomous():
			plpy.execute("insert into atx_test values ('bsd', 456)")
	except (plpy.SPIError, e):
		print("error: %s" % e.args)

	plpy.execute("insert into atx_test values ('csd', 'csd')")
$$ LANGUAGE plpythonu;

select pythonomous();
select * from atx_test; -- you should see (bsd, 456)

drop table atx_test;
