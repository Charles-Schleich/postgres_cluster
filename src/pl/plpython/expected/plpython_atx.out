CREATE EXTENSION plpythonu;
create table atx_test(a text, b int);
create or replace function pythonomous() returns void as $$
	plpy.execute("insert into atx_test values ('asd', 123)")

	try:
		with plpy.autonomous():
			plpy.execute("insert into atx_test values ('bsd', 456)")
	except (plpy.SPIError, e):
		print("error: %s" % e.args)

	plpy.execute("insert into atx_test values ('csd', 'csd')")
$$ LANGUAGE plpythonu;
select pythonomous();
ERROR:  spiexceptions.InvalidTextRepresentation: invalid input syntax for integer: "csd"
LINE 1: insert into atx_test values ('csd', 'csd')
                                            ^
QUERY:  insert into atx_test values ('csd', 'csd')
CONTEXT:  Traceback (most recent call last):
  PL/Python function "pythonomous", line 10, in <module>
    plpy.execute("insert into atx_test values ('csd', 'csd')")
PL/Python function "pythonomous"
select * from atx_test; -- you should see (bsd, 456)
  a  |  b  
-----+-----
 bsd | 456
(1 row)

drop table atx_test;
